### PR TITLE
V15: Cache seeding docs

### PR DESCRIPTION
**Pull request by nikolajlauridsen**
## Description

* Added a guide for creating custom seed key providers
* Added general information about cache seeding
* Updated nucachesettings.md to cache-settings.md and added relevant configuration information for the new cache

## Type of suggestion

* [ ] Typo/grammar fix
* [ ] Updated outdated content
* [X] New content
* [X] Updates related to a new version
* [ ] Other

## Product & version (if relevant)
CMS V15

_When should the content be published?_

When V15-RC3 is released (The articles contains information about configuration that's not there until then)

### CODE REVIEW

File: 15/umbraco-cms/extending/creating-custom-seed-key-provider.md
Comment (sofietoft) [inline 15/umbraco-cms/extending/creating-custom-seed-key-provider.md:11]: I suggest we remove the version number. I mean, is it relevant to this article? 🤔 You'll already be viewing the docs for version 15 😅 

```suggestion
Umbraco uses a lazy loaded cache, which means that content is loaded into the cache on an as-needed basis. However, you may need specific content to always be in the cache. To achieve this you can implement your own custom seed key providers.

There are two types of seed key providers: `IDocumentSeedKeyProvider` for documents and `IMediaSeedKeyProvider` for media. As these interfaces are identical only `IDocumentSeedKeyProvider` is demonstrated in this article.
```

File: 15/umbraco-cms/extending/creating-custom-seed-key-provider.md
Comment (sofietoft) [inline 15/umbraco-cms/extending/creating-custom-seed-key-provider.md:14]:
```suggestion
Seed keys are cached and calculated once. Any documents created after the site has started will not be included in the seed keys until after a server restart.
```

File: 15/umbraco-cms/extending/creating-custom-seed-key-provider.md
Comment (sofietoft) [inline 15/umbraco-cms/extending/creating-custom-seed-key-provider.md:21]:
```suggestion
1. Create a new class called `BlogSeedKeyProvider` that implements `IDocumentSeedKeyProvider`.
```

File: 15/umbraco-cms/extending/creating-custom-seed-key-provider.md
Comment (sofietoft) [inline 15/umbraco-cms/extending/creating-custom-seed-key-provider.md:54]:
```suggestion
3. Parse a hardcoded string to a GUID.
4. Use the `IDocumentNavigationQueryService` to get the children of the blog node.
5. Return their keys as a `HashSet`.
```

File: 15/umbraco-cms/extending/creating-custom-seed-key-provider.md
Comment (sofietoft) [inline 15/umbraco-cms/extending/creating-custom-seed-key-provider.md:69]:
```suggestion
Since this returns it as a set, and all the sets get unioned, we do not have to worry about duplicates.
```

File: 15/umbraco-cms/extending/creating-custom-seed-key-provider.md
Comment (sofietoft) [inline 15/umbraco-cms/extending/creating-custom-seed-key-provider.md:102]:
```suggestion
Now that the `BlogSeedKeyProvider` is implemented, it must be registered in the `Startup` class.
```

File: 15/umbraco-cms/extending/creating-custom-seed-key-provider.md
Comment (sofietoft) [inline 15/umbraco-cms/extending/creating-custom-seed-key-provider.md:115]:
```suggestion
All blogpost will now be seeded into the cache on startup, and will always be present in the cache.
```

File: 15/umbraco-cms/reference/cache/cache-seeding.md
Comment (sofietoft) [inline 15/umbraco-cms/reference/cache/cache-seeding.md:8]:
```suggestion
Umbraco uses a lazy loaded cache, meaning content is loaded into the cache on an as-needed basis. Whenever a piece of content is shown on the website for the first time it first needs to be loaded into the cache.
```

File: 15/umbraco-cms/reference/cache/cache-seeding.md
Comment (sofietoft) [inline 15/umbraco-cms/reference/cache/cache-seeding.md:11]:
```suggestion
For certain pages, like the front page, you may not want this delay to be there. The role of cache seeding is meant to solve this issue.
```

**Review by sofietoft — CHANGES_REQUESTED** [review]
There were quite a few too long sentences in the articles here, so I took the liberty to make changes directly to the PR 🙈 felt that was a bit quicker.
Please give it a look and let me know if something doesn't look right! 🙌 

Besides grammar, readability, and typos, I've also made some suggestions.
For the Extending article, I suggest we make that into a step-by-step guide! 🤞 

Finally, three new articles are added with this PR and one is removed.
This means we need to do a few more things here:

* Add the new articles to the SUMMARY file
* Add a redirect for the deleted article (I suggest we redirect to the new cache-settings.md article, and add a note at the top of that article to explain why the nucachesettings article is done.

Let me know if you prefer that I handle added the redirect and the SUMMARY entries - I'd be happy to! 🙌

File: 15/umbraco-cms/extending/creating-custom-seed-key-provider.md
Comment (nikolajlauridsen) [reply 15/umbraco-cms/extending/creating-custom-seed-key-provider.md]: No that makes a lot of sense 😄

**Comment by nikolajlauridsen** [comment]
Thank you 🙌 Brevity is not my forté 🙈 not sure how the whole redirect thing works, so would be great if you can handle that 😄

**Comment by sofietoft** [comment]
I've added the new articles to the SUMMARY file and made the redirects.

Before I merge, can you help me confirm whether it's OK to redirect the NuCache settings article to the new Cache Settings article? 🤔 Or should I redirect it somewhere else?

**Comment by nikolajlauridsen** [comment]
Awesome, thanks 🙌

Yes I think it makes a lot of sense to redirect it to the new cache settings article, since NuCache no longer exists, and has been replaced by the new cache 😄

**Comment by sofietoft** [comment]
Great! Actually, now that I think about it - I'll add a little note to the new article.
Just in case someone comes looking for the NuCache article.

**Comment by sofietoft** [comment]
@nikolajlauridsen - When you have a spare minute, could you take a look at this commit: https://github.com/umbraco/UmbracoDocs/pull/6578/commits/12c0fc83fd11a61e94abbf8d59edb5d7bb5c75e9 🤞 

Just want to make sure I got it right!

**Comment by nikolajlauridsen** [comment]
That looks perfect to me 👌

**Comment by sofietoft** [comment]
Excellent! Then I think this is finally ready 💪 

You're releasing RC-3 next week, right?

**Comment by nikolajlauridsen** [comment]
Yeps, it's targeted for next week 😄

**Comment by nikolajlauridsen** [comment]
Also, this is a bit odd, but I've had to extent the article with some additional information, not sure what's the easiest workflow for you, but I've made a separate PR here https://github.com/nikolajlauridsen/UmbracoDocs/pull/1

Migrating this will update this PR, but I thought it might be easier to only review the new changes alone, but since I don't have commit rights to this repo it had to be on my own, please let me know if I need to do something different 😄

**Review by sofietoft — COMMENTED** [review]
New changes looks good @nikolajlauridsen !

I've added a suggestion for a small rewrite - let me know what you think!

File: 15/umbraco-cms/reference/configuration/cache-settings.md
Comment (nikolajlauridsen) [reply 15/umbraco-cms/reference/configuration/cache-settings.md:21]: That looks great! Thanks 😄

**Pull request merged**
